README.Rmd

```diff
@@ -609,12 +609,15 @@ The code in this repository is written primarily in [R](https://www.r-project.or
 
 If you're on Windows, you'll also need to install [Rtools](https://cran.r-project.org/bin/windows/Rtools/) in order to build the necessary packages. You may also want to (optionally) install [DVC](https://dvc.org/doc/install) to pull data and run pipelines.
 
+We also publish a Docker image containing model code and all of the dependencies necessary to run it. If you're comfortable using Docker, you can skip the installation steps below and instead pull the image from `ghcr.io/ccao-data/model-res-avm:master` to run the model.
+
 ## Installation
 
 1. Clone this repository using git, or simply download it using the button at the top of the page.
 2. Set your working directory to the local folder containing this repository's files, either using R's `setwd()` command or (preferably) using RStudio's [projects](https://support.posit.co/hc/en-us/articles/200526207-Using-Projects).
 3. Install `renv`, R's package manager, by running `install.packages("renv")`.
 4. Install all R package dependencies using `renv` by running `renv::restore()`. This step may take awhile. Linux users will likely need to install dependencies (via apt, yum, etc.) to build from source.
+   1. The `finalize` step of the model pipeline requires some additional dependencies for generating a model performance report. These dependencies must be installed in addition to the core dependnecies installed in step 4. If you would like to run this step, make sure to install its additional dependencies by running `renv::restore(lockfile = "renv/profiles/reporting/renv.lock")`.
```

Comment:
I'm uncertain whether or not we want to communicate that reporting dependencies are required, and by extension whether `finalize` is a required step. On the one hand, the isolation of its dependencies implies to me that it should be an optional step, since the point of the dependency isolation is to try to narrow down the list of dependencies required for running the core model; on the other hand, `finalize` is part of the pipeline by default, so things won't really fail gracefully if a user neglects to install reporting dependencies but then runs `dvc repro`. My thinking is that it might be cleaner to update `05-finalize.R` to be a bit smarter, i.e. to detect whether optional dependencies are installed and only generate the report if so, but I'll leave some more thoughts on that in the source code below.

Reply:
That makes sense! I decided to split `finalize` into two workflows, one that generates stats and reports (`05-finalize.R`) and another that uploads them to S3 (`06-upload.R`). I also updated the docs to reflect this change and clarify that external users can run `finalize` but not `upload`.